src/mococrw/ecies.h

```diff
@@ -0,0 +1,332 @@
+     * @return builder instance
+     */
+    ECIESCtxBuilder &setSymmetricCipherKeySize(mococrw::SymmetricCipherKeySize keySize);
+
+    /**
+     * @brief Sets the padding for the symmetric encryption. (Default: mococrw::SymmetricCipherPadding::NO)
+     * @param padding The padding scheme to be used.
+     * @return builder instance
+     */
+    ECIESCtxBuilder &setSymmetircCipherPadding(mococrw::SymmetricCipherPadding padding);
+
+    /**
+     * @brief Sets the initialisation vector (IV) for the symmetric encryption. (Default: zeros)
+     * @param iv The initialisation vector. The length shall be the same as the key length
+     * @return builder instance
+     */
+    ECIESCtxBuilder &setSymmetricCipherIv(std::vector<uint8_t> iv);
+
+    /**
+     * @brief Sets the salt for the key derivation function (KDF). (Default: empty vector -> not used)
+     * @param kdfSalt The salt value used for key derivation.
+     * @return builder instance
+     */
+    ECIESCtxBuilder &setKDFSalt(std::vector<uint8_t> kdfSalt);
+
+    /**
+     * @brief Sets the salt for the message authentication code. (Default: empty vector -> not used).
+     * @param macSalt The salt value used for message authentication code.
+     * @return builder instance
+     */
+    ECIESCtxBuilder &setMACSalt(std::vector<uint8_t> macSalt);
+
+protected:
+    /**
+     * @brief constructor
+     *
+     * This class shall not be instantiated (except by subclasses)
+     */
+    ECIESCtxBuilder();
+
+private:
+    std::shared_ptr<mococrw::KeyDerivationFunction> kdf;
+    std::shared_ptr<mococrw::MessageAuthenticationCode> mac;
+    mococrw::SymmetricCipherMode mode;
+    mococrw::SymmetricCipherKeySize keySize;
+    mococrw::SymmetricCipherPadding padding;
+    std::vector<uint8_t> iv;
+    std::vector<uint8_t> kdfSalt;
+    std::vector<uint8_t> macSalt;
+
+};
+
+class ECIESEncryptionCtxBuilder : public ECIESCtxBuilder
+{
+public:
+    /**
+     * \brief Constructor for public keys
+     * \param bobsKey The public key of the receiver.
+     */
+    ECIESEncryptionCtxBuilder(const AsymmetricPublicKey& bobsKey);
+
+    /**
+     * @brief Constructor if the public key is inside a certificate
+     * @param bobsCert The certificate of the receiver which contains his/her public key.
+     */
+    ECIESEncryptionCtxBuilder(const X509Certificate& bobsCert);
+
+    /**
+     * @brief Create the ECIES encryption context.
+     *
+     * Call this function after you have set the options, which differ from the default value
+     * @return An ECIES encryption context, which can be used for encryption.
+     */
+    std::unique_ptr<ECIESEncryptionCtx> buildEncryptionCtx();
+
+private:
+    AsymmetricPublicKey pubKey;
+    X509Certificate cert;
+};
+
+class ECIESDecryptionCtxBuilder : public ECIESCtxBuilder
+{
+public:
+    /**
+     * @brief Constructor
+     * @param bobsKey The private key of the receiver. This key is needed for decrypting the ciphertext
```

Comment:
```suggestion
     * @param bobsKey The receiver's private key. This key is needed for decrypting the ciphertext
```
My English teacher didn't stop mentioning this.